src/Azure/Azure.Quantum.Client/JobManagement/Workspace.cs

```diff
@@ -267,5 +313,8 @@ public async Task<string> GetSasUriAsync(string containerName, string blobName =
                 jobId,
                 inner);
         }
+
+        private static string NormalizeLocation(string location) =>
+            string.Concat(location.Where(c => !char.IsWhiteSpace(c))).ToLower();
```

Comment:
In IQ#, we also check that the resulting string is a valid hostname. Otherwise the `Uri` that is created may be invalid or point to some totally different domain. (e.g., imagine the user provides `mydomain.com/quantum` as a `--location` parameter.) See here:
https://github.com/microsoft/iqsharp/blob/a31142b969f4c49b82338c3ea61b8edbde339d52/src/AzureClient/AzureEnvironment.cs#L73-L78
Obviously this is purely a usability thing, not a security thing, since the user already has the ability to override the URI via `--base-uri`. Up to you whether you want to do that.

Reply:
That's a good point, thanks for the suggestion, I'll incorporate it :)

Reply:
This verification is now done through a validator in the command line option itself.